omero/util_scripts/Min_Max.py

```diff
@@ -0,0 +1,259 @@
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+------------------------------------------------------------------------------
+"""
+
+from omero.gateway import BlitzGateway
+from omero.model import StatsInfoI
+from omero.rtypes import rdouble, rlong, rstring
+from omero.util.text import TableBuilder
+from omero.util.tiles import TileLoop, TileLoopIteration
+
+import omero.scripts as scripts
+import omero.util.script_utils as script_utils
+
+from collections import defaultdict
+from random import shuffle
+
+from numpy import amin, amax, iinfo
+from numpy import average as avg
+
+
+def calcStatsInfo(conn, imageId, choice, debug=False):
+    """
+    Process a single image here: creating a new StatsInfo object
+    if necessary.
+
+    @param imageId:             Original image
+    """
+
+    oldImage = conn.getObject("Image", imageId)
+    if oldImage is None:
+        raise Exception("Image not found for ID:" % imageId)
+
+    sizeX = oldImage.getSizeX()
+    sizeY = oldImage.getSizeY()
+    sizeZ = oldImage.getSizeZ()
+    sizeC = oldImage.getSizeC()
+    sizeT = oldImage.getSizeT()
+    tileW = min(256, sizeX)
+    tileH = min(256, sizeY)
+
+    zctMap = defaultdict(list)
+
+    class Loop(TileLoop):
+
+        def createData(self):
+            return self
+
+        def close(self):
+            pass
+
+    only_default = (choice == "default")
+
+    class Iteration(TileLoopIteration):
+
+        def run(self, data, z, c, t, x, y,
+                tileWidth, tileHeight, tileCount):
+
+            if only_default:
+                if t != 0 or z != int(sizeZ/2):
+                    return
+            zctMap[c].append(
+                (z, c, t, (x, y, tileWidth, tileHeight)))
+
+    Loop().forEachTile(
+        sizeX, sizeY,
+        sizeZ, sizeC, sizeT,
+        tileW, tileH, Iteration())
+
+    if choice == "random":
+        for c in zctMap:
+            copy = list(zctMap[c])
+            if len(copy) >= 100:
+                copy = copy[0:100]
+            shuffle(copy)
+            zctMap[c] = copy
+
+    def channelGen():
+        byte_count = 0
+        tile_count = 0
+        pixels = oldImage.getPrimaryPixels()
+        rv = dict()
+        dt = pixels.getTile(0, 0, 0, (0, 0, 16, 16)).dtype
+        tile_min = iinfo(dt).max  # Everything is less
+        tile_max = iinfo(dt).min  # Everything is more
+        for c, zctTileList in zctMap.items():
+            for tileInfo in zctTileList:
+                tile = pixels.getTile(*tileInfo)
+                tile_min = min(tile_min, amin(tile))
+                tile_max = max(tile_max, amax(tile))
+                byte_count += tile.nbytes
+                tile_count += 1
+            rv[c] = (tile_min, tile_max)
+        yield rv, byte_count, tile_count
+
+    statsInfos = dict()
+    for x, byte_count, tile_count in channelGen():
+        statsInfos.update(x)
+
+    if debug:
+        print "Loaded %s tile(s) (%s bytes)" % (tile_count, byte_count)
```

Comment:
This one and other similar print statements error in my hands. Is it because of the python3 environment I am using ?

Reply:
Yes, just use `print()` like 
```
print("Loaded %s tile(s) (%s bytes)" % (tile_count, byte_count))
```